### ISSUE TITLE
Numerical checks failing in pre-release builds

### ISSUE THREAD

**Issue by effigies** [nipy/nibabel#968]
Example: https://travis-ci.com/github/nipy/nibabel/jobs/434783290

```
    @needs_scipy
    def test_spatial_axes_check(caplog):
        for fname in MINC_3DS + OTHER_IMGS:
            img = nib.load(pjoin(DATA_DIR, fname))
            with caplog.at_level(logging.CRITICAL):  # Suppress logs when changing classes
                s_img = smooth_image(img, 0)
            assert_array_equal(img.dataobj, s_img.dataobj)
            with caplog.at_level(logging.CRITICAL):
                out = resample_from_to(img, img, mode='nearest')
>           assert_almost_equal(img.dataobj, out.dataobj)
E           AssertionError:
E           Arrays are not almost equal to 7 decimals
E
E           Mismatched elements: 7140 / 21420 (33.3%)
E           Max absolute difference: 0.78091769
E           Max relative difference: 0.00083934
E            x: array([[[[4004.1372025, 4039.7292917, 4020.1234798, ..., 3986.4919719,
E                     3994.5605175, 3971.1089503],
E                    [3697.2308403, 3726.6395581, 3703.4142118, ..., 3697.5324682,...
E            y: array([[[[4003.6113328, 4039.1997538, 4019.5976086, ..., 3985.9655399,
E                     3994.03126  , 3970.5845459],
E                    [3697.2308403, 3726.6395581, 3703.4142118, ..., 3697.5324682,...
```

First failures showed up about ~21 hours, so starting around November 12 @ 7pm UTC (2pm EST).

Have a successful build 3 days ago, and scipy has a nightly wheel out 2.5 days ago.

25.1 MB | \|                              scipy-1.6.0.dev0+20201111022934_2af2d98-cp38-cp38-manylinux1_x86_64.whl | 2 days and 13 hours ago | scipy-wheels-nightly | 253 |  
-- | -- | -- | -- | -- | --

That's my best guess, but I probably won't have time to track down more details until Monday.

@grlee77 @larsoner Would either of you care to have a look?

**Comment by larsoner** [comment]
This does look like it's likely due to one of your recent PRs @grlee77 . Can you look?

Glad you tests the `--pre` so we found this before 1.6.0 comes out in a couple of weeks!

**Comment by grlee77** [comment]
yes, I will take a look today

**Comment by grlee77** [comment]
I cannot reproduce locally with current SciPy master. I suspect this was fixed by the improvements to modes 'nearest' and 'constant' that were merged ~14 hours ago in scipy/scipy#12776.

If it occurs again on a newer nightly build, let's revisit the issue

**Comment by effigies** [comment]
I don't follow scipy comms closely, so not sure if they're aware that the latest nightly build only pushed 4 wheels? https://anaconda.org/scipy-wheels-nightly/scipy/files
